core/domain/platform_feature_services.py

```diff
@@ -149,21 +150,21 @@ def update_feature_flag_rules(
         feature_name, committer_id, commit_message, new_rules)
 
 
-# TODO(#10211): Currently Oppia runs in either of the two modes:
-# dev or prod. There should be another mode 'test' added for QA testing,
-# once it is added, this function needs to be updated to take that into
-# consideration.
 def _get_server_mode() -> platform_parameter_domain.ServerMode:
     """Returns the running mode of Oppia.
 
     Returns:
-        Enum(SERVER_MODES). The server mode of Oppia, dev if Oppia is running
-        in development mode, prod if in production mode.
+        Enum(SERVER_MODES). The server mode of Oppia. This is "dev" if Oppia is
+        running in development mode, "test" if Oppia is running in production
+        mode but not on the main website, and "prod" if Oppia is running in
+        full production mode on the main website.
     """
     return (
         platform_parameter_domain.ServerMode.DEV
         if constants.DEV_MODE
         else platform_parameter_domain.ServerMode.PROD
+        if feconf.ENV_IS_OPPIA_ORG_PRODUCTION_SERVER
+        else platform_parameter_domain.ServerMode.TEST
     )
```

Comment:
Nit: Should we simplify the nested ternary operator for simplicity?

Reply:
After the offline discussion, it doesn't matter since we won't be adding more cases to this and the docstring explains the function.

resolving the comment